consensus/src/pow/constants.rs

```diff
@@ -0,0 +1,14 @@
+use common::Uint256;
+
+pub const TARGET_TIMESPAN_SECS: u32 = 14 * 24 * 60 * 60; // 2 weeks
```

Comment:
This whole file should probably go into ChainConfig.

- TARGET_TIMESPAN_SECS should be in ChainConfig, and should be returned as rust Duration.
- Uint256 version should be derived from that, and not hardcoded.
- TARGET_SPACING should certainly be in ChainConfig.
- DIFFICULTY_ADJUSTMENT_INTERVAL should be a function.
- TIMESPAN_ADJUSTMENT_FACTOR: This should be called MAX_DIFFICULTY_ADJUSTMENT_PER_INTERVAL, which is basically the max increase/decrease in work that can happen. This protects from having huge swings in difficulty. Also should go into ChainConfig
- Finally, the upper and lower things also should be functions.

Preferably, find a way to group them all under PoW in ChainConfig instead of having them scattered. Maybe just prefix? Maybe substruct? I don't know. Please propose something.